### PR TITLE
[ClassLoader] Fix parsing namespace when token_get_all() is missing

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14871 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Not bullet proof, but that's not a target (if you want a bullet proof solution, enable token_get_all).

**Comment by stof** [comment]
:+1:

**Comment by Tobion** [comment]
:+1:

**Comment by Tobion** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/ClassLoader/ClassCollectionLoader.php [hunk 1]
```diff
@@ -137,8 +137,8 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
     public static function fixNamespaceDeclarations($source)
     {
         if (!function_exists('token_get_all') || !self::$useTokenizer) {
-            if (preg_match('/namespace(.*?)\s*;/', $source)) {
-                $source = preg_replace('/namespace(.*?)\s*;/', "namespace$1\n{", $source)."}\n";
+            if (preg_match('/(^|\s)namespace(.*?)\s*;/', $source)) {
```
Comment (WouterJ) [inline src/Symfony/Component/ClassLoader/ClassCollectionLoader.php]: Isn't `\b` in regexes meant for this?

Comment (nicolas-grekas) [inline src/Symfony/Component/ClassLoader/ClassCollectionLoader.php]: \b would match e.g. `:namespace`

Comment (WouterJ) [inline src/Symfony/Component/ClassLoader/ClassCollectionLoader.php]: Ah, thanks!

**Pull request merged**